plugins/modules/ndo_physical_interface.py

```diff
@@ -0,0 +1,404 @@
+            name=dict(type="str", aliases=["physical_interface"]),
+            uuid=dict(type="str", aliases=["physical_interface_uuid"]),
+            description=dict(type="str"),
+            nodes=dict(type="list", elements="int"),
+            interfaces=dict(type="list", elements="str"),
+            physical_interface_type=dict(type="str", choices=["physical", "breakout"]),
+            physical_policy_uuid=dict(type="str", aliases=["policy_uuid", "interface_policy_uuid", "interface_policy_group_uuid", "interface_setting_uuid"]),
+            physical_policy=dict(
+                type="dict",
+                options=dict(
+                    name=dict(type="str"),
+                    template=dict(type="str"),
+                ),
+                aliases=["policy", "interface_policy", "interface_policy_group", "interface_setting"],
+            ),
+            breakout_mode=dict(type="str", choices=["4x10G", "4x25G", "4x100G"]),
+            interface_descriptions=dict(
+                type="list",
+                elements="dict",
+                options=dict(
+                    interface_id=dict(type="str"),
+                    description=dict(type="str"),
+                ),
+            ),
+            state=dict(type="str", default="query", choices=["absent", "query", "present"]),
+        )
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "present", ["name", "uuid"], True],
+            ["state", "absent", ["name", "uuid"], True],
+        ],
+        mutually_exclusive=[("physical_policy", "breakout_mode"), ("physical_policy", "physical_policy_uuid")],
+    )
+
+    mso = MSOModule(module)
+
+    template = module.params.get("template")
+    name = module.params.get("name")
+    uuid = module.params.get("uuid")
+    description = module.params.get("description")
+    nodes = module.params.get("nodes")
+    if nodes:
+        nodes = [str(node) for node in nodes]
+    interfaces = module.params.get("interfaces")
+    if interfaces:
+        interfaces = ",".join(interfaces)
+    physical_interface_type = module.params.get("physical_interface_type")
+    physical_policy_uuid = module.params.get("physical_policy_uuid")
+    physical_policy = module.params.get("physical_policy")
+    breakout_mode = module.params.get("breakout_mode")
+    interface_descriptions = module.params.get("interface_descriptions")
+    state = module.params.get("state")
+
+    ops = []
+    match = None
+
+    mso_template = MSOTemplate(mso, "fabric_resource", template)
+    mso_template.validate_template("fabricResource")
+
+    object_description = "Physical Interface Profile"
+
+    existing_physical_interfaces = mso_template.template.get("fabricResourceTemplate", {}).get("template", {}).get("interfaceProfiles", [])
+
+    if state in ["query", "absent"] and existing_physical_interfaces == []:
```

Comment:
interfaceProfiles return null when empty, this is deviation is API
likely you want to do an falsified check here
```suggestion
    if state in ["query", "absent"] and not existing_physical_interfaces:
```